nomos-services/network/src/backends/libp2p/mixnet.rs

```diff
@@ -47,11 +47,22 @@ impl MixnetHandler {
     }
 
     pub async fn run(&mut self) {
-        let Ok(mut stream) = self.client.run().await else {
-            tracing::error!("Could not quickstart mixnet stream");
-            return;
-        };
+        // we need this loop to help us reestablish the connection in case
+        // the mixnet client fails for whatever reason
+        loop {
+            tokio::select! {
```

Comment:
why as select if there's only one branch?

Reply:
Ah, it is a left-over.